### PR TITLE
Correct semantics for link opening in global search

### LINKED ISSUE #41800: Use href & target attribute for NcListItem in search result
### Bug description

Use href of `NcListItem` once https://github.com/nextcloud-libraries/nextcloud-vue/issues/4880 is completed.

### PR DESCRIPTION
**Pull request by nfebe**
Following the update of `NcListItem` to allow control over where and how passed `hrefs` open, the search result component can now make use of these props.

See : https://github.com/nextcloud-libraries/nextcloud-vue/pull/4923

Resolves :  #41800

### CODE REVIEW

**Review by artonge — APPROVED** [review]

File: core/src/components/GlobalSearch/SearchResult.vue [hunk 1]
```diff
@@ -2,7 +2,8 @@
 	<NcListItem class="result-items__item"
 		:name="title"
 		:bold="false"
-		@click="openResult(resourceUrl)">
+		:href="resourceUrl"
+		:target="'_self'">
```
Comment (susnux) [inline core/src/components/GlobalSearch/SearchResult.vue:6]:
```suggestion
		target="_self">
```

**Review by susnux — APPROVED** [review]

**Comment by Fenn-CS** [comment]
Everything green except `sqlite-php8.0-samba-native` which is unrelated.

**Pull request merged**